### PR TITLE
add gRPC Sink and Source

### PR DESCRIPTION
**Pull request by niruhan**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

* O/S: MacOS Mojave 10.14.3
* Java version: 1.8.0.201

### CODE REVIEW

File: component/src/main/java/io/siddhi/extension/io/grpc/sink/GRPCSink.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+            if (payload instanceof SequenceCallRequest) {
+                if (methodName.equalsIgnoreCase("CallSequenceWithResponse")) {
+                    ListenableFuture<SequenceCallResponse> futureResponse =
+                            futureStub.callSequenceWithResponse((SequenceCallRequest) payload);
+                    Futures.addCallback(futureResponse, new FutureCallback<SequenceCallResponse>() {
+                        @Override
+                        public void onSuccess(SequenceCallResponse result) {
+                            sourceStaticHolder.getGRPCSource(sinkID).onResponse(result);
+                            if (logger.isDebugEnabled()) {
+                                logger.debug("Success!");
```
Comment (menuka94) [inline component/src/main/java/io/siddhi/extension/io/grpc/sink/GRPCSink.java]: Let's add more elaborate debug logs.

File: component/src/main/java/io/siddhi/extension/io/grpc/sink/GRPCSink.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+                            }
+                        }
+
+                        @Override
+                        public void onFailure(Throwable t) {
+                            if (logger.isDebugEnabled()) {
+                                logger.debug("Failure!");
```
Comment (menuka94) [inline component/src/main/java/io/siddhi/extension/io/grpc/sink/GRPCSink.java]: Ditto.

File: .gitignore [hunk 3]
```diff
@@ -1,23 +1,27 @@
...
-*.nar
 *.ear
-*.zip
-*.tar.gz
-*.rar
 
 # virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
 hs_err_pid*
+
+#idea files
```
Comment (madushadhanushka) [inline .gitignore]: Why we are adding .idea, .DS_Store etc into the gitignore file?

**Comment by AnuGayan** [comment]
License header is missing in some files. Please check..

File: component/pom.xml [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+        </dependency>
+        <dependency>
+            <groupId>io.siddhi.extension.map.json</groupId>
+            <artifactId>siddhi-map-json</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.grpc</groupId>
+            <artifactId>grpc-netty-shaded</artifactId>
+            <version>1.21.0</version>
```
Comment (AnuGayan) [inline component/pom.xml]: Move versions to the parent pom

File: component/pom.xml [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Export-Package>
+                            io.siddhi.extension.io.http.*
```
Comment (AnuGayan) [inline component/pom.xml]: Why we export io.http here? It seems like there are two bundle plugin configurations in this [pom](https://github.com/siddhi-io/siddhi-io-grpc/pull/1/files#diff-fc07a885254925c2e683437749faa0f8R190)

**Comment by niruhan** [comment]
closing as some architectural changes are required and this is not the final cut

**Pull request closed without merging**